aspen/website.py

```diff
@@ -68,15 +69,19 @@ def redirect(self, location, code=None, permanent=False, base_url=None, response
         permanent is True and 302 (Found) if it is False. If url doesn't start
         with base_url (defaulting to self.base_url), then we prefix it with
         base_url before redirecting. This is a protection against open
-        redirects. If you provide your own response we will set .code and
-        .headers['Location'] on it.
+        redirects. If base_url is not '' and location is not an absolute path,
```

Comment:
How about "If location is a relative path but base_url is empty, we raise BadLocation."